stacks-common/build.rs

```diff
@@ -1,8 +1,64 @@
 use std::path::Path;
+use std::process::Command;
 use std::{env, fs};
 
 use toml::Value;
 
+fn current_git_hash() -> Option<String> {
+    if option_env!("GIT_COMMIT") == None {
+        let commit = Command::new("git")
+            .arg("log")
+            .arg("-1")
+            .arg("--pretty=format:%h") // Abbreviated commit hash
+            .current_dir(env!("CARGO_MANIFEST_DIR"))
+            .output();
+
+        if let Ok(commit) = commit {
+            if let Ok(commit) = String::from_utf8(commit.stdout) {
+                return Some(commit.trim().to_string());
+            }
+        }
+    } else {
+        return option_env!("GIT_COMMIT").map(String::from);
```

Comment:
It seems like there should be a Rustier way to write these, with `or_else` maybe? After pulling the git command into a function, it may be more obvious.

Reply:
I agree with Brice here. Here's my suggestion:

```rust
fn current_git_hash() -> Option<String> {
    option_env!("GIT_COMMIT")
        .map(String::from)
        .or_else(|| {
            Command::new("git")
                .arg("log")
                .arg("-1")
                .arg("--pretty=format:%h") // Abbreviated commit hash
                .current_dir(env!("CARGO_MANIFEST_DIR"))
                .output()
                .map(|output| String::from_utf8(output.stdout))
                .ok()
        })
}
```

You can do something similar for `current_git_branch()` also